src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -34,5 +34,28 @@ internal class ParsedCallExpression
         public IEnumerable<ParsedArgumentExpression> ArgumentsExpressions { get; }
 
         public object CallTarget => this.callTarget.Value;
+
+        // Expression evaluation optimized for a very common case: 
+        // - it's a 'MemberAccess' expression on a local lambda closure (A.CallTo(() => fakeObj.doSomething(args));

```

Comment:
@blairconrad I'm not sure I understand your comments. From what I can tell, the goal is to get the value of `fakeObj` from `A.CallTo(() => fakeObj.doSomething(args)`